xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/internal/objects/classes/ImplicitlyAllowedValuesDBListQueryBuilder.java

```diff
@@ -156,8 +174,17 @@ private void addFieldToQuery(String fieldName, String fieldAlias, boolean hasCla
         } else if (!hasClassName) {
             selectClause.add(DOC_PREFIX + fieldName);
         } else {
+            PropertyClass pc = null;
+            try {
+                pc = (PropertyClass) contextProvider.get().getWiki()
+                    .getXClass(documentReferenceResolver.resolve((String) parameters.get(CLASS_NAME)),
+                        contextProvider.get()).get(fieldName);
+            } catch (XWikiException e) {
+                logger.error("Failed to get property [{}] for class [{}]", fieldName, parameters.get(CLASS_NAME), e);
```

Comment:
If you can continue after such an exception (I'm not sure you can) then you should log an warning message instead, with the root cause message. Otherwise you should simply throw an exception. See also https://dev.xwiki.org/xwiki/bin/view/Community/CodeStyle/JavaCodeStyle/#HLoggingBestPractices .